src/main/kotlin/org/jitsi/nlj/stats/DelayStats.kt

```diff
@@ -16,33 +16,115 @@
 
 package org.jitsi.nlj.stats
 
-import java.util.Arrays
 import java.util.concurrent.atomic.AtomicLong
 import java.util.concurrent.atomic.LongAdder
 import org.jitsi.nlj.PacketInfo
+import org.jitsi.nlj.util.OrderedJsonObject
+import org.jitsi.utils.increaseAndGet
 
-open class DelayStats {
+open class DelayStats(
+    thresholdsNoMax: LongArray = longArrayOf(2, 5, 20, 50, 200, 500, 100)
+) {
     private val totalDelayMs = LongAdder()
     private val totalCount = LongAdder()
-    val averageDelay: Double
+    private val averageDelayMs: Double
         get() = totalDelayMs.sum() / totalCount.sum().toDouble()
-    val maxDelayMs = AtomicLong(0)
+    private val maxDelayMs = AtomicLong(0)
 
-    /* TODO: make thresholds configurable */
-    val thresholds = longArrayOf(2, 5, 20, 50, 200, 500, 1000)
-    val thresholdCounts = Array(thresholds.size + 1) { LongAdder() }
+    private val thresholds = longArrayOf(*thresholdsNoMax, Long.MAX_VALUE)
+    private val thresholdCounts = Array(thresholds.size + 1) { LongAdder() }
 
     fun addDelay(delayMs: Long) {
         if (delayMs >= 0) {
             totalDelayMs.add(delayMs)
-            if (delayMs > maxDelayMs.get()) {
-                maxDelayMs.set(delayMs)
-            }
+            maxDelayMs.increaseAndGet(delayMs)
             totalCount.increment()
 
-            val searchResult = Arrays.binarySearch(thresholds, delayMs)
-            val bucket = if (searchResult < 0) { - searchResult - 1 } else { searchResult }
-            thresholdCounts[bucket].increment()
+            findBucket(delayMs).increment()
+        }
+    }
+
+    private fun findBucket(delayMs: Long): LongAdder {
+        // The vast majority of values are in the first bucket, so linear search is likely faster than binary.
+        for (i in thresholds.indices) {
+            if (delayMs <= thresholds[i]) return thresholdCounts[i]
+        }
+        return thresholdCounts.last()
+    }
```

Comment:
Why do you need both `Long.MAX_VALUE` in `thresholds` and the default return here?

Reply:
Because the compiler doesn't know the last threshold is always `MAX_VALUE` and the loop never completes, so it requires a return statement.

Reply:
I think a `when` might accomplish the same thing here?

Reply:
But if you have to have the loop termination statement anyway, you don't need the `MAX_VALUE`.  Which I think means you don't need to copy the `thresholds` array (though you may want a local copy to make sure your caller doesn't change it out from under you).

Reply:
(Though maybe you still need it for the p99 code?)

Reply:
We need an extra bucket for "larger than the last threshold" anyway, so it's just a matter of how it's described. This way it is easier to handle in the rest of the code because it's a valid `(threshold, counter)` pair.